pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -237,20 +241,27 @@ def _grouped_plot_by_column(
         columns = data._get_numeric_data().columns.difference(by)
     naxes = len(columns)
     fig, axes = create_subplots(
-        naxes=naxes, sharex=True, sharey=True, figsize=figsize, ax=ax, layout=layout
+        naxes=naxes, sharex=kwargs.pop("sharex", True),
+        sharey=kwargs.pop("sharey", True), figsize=figsize, ax=ax, layout=layout
     )
 
     _axes = flatten_axes(axes)
 
+    # GH 45465: move the "by" label based on "vert"
+    xlabel, ylabel = kwargs.pop("xlabel", None), kwargs.pop("ylabel", None)
+    if kwargs.get("vert", 1):
```

Comment:
Nit: Why 1 instead of True?

Reply:
That was how it was originally in the code before my edits. Happy to change it though 🙂

Reply:
Gotcha. If you could, thanks!

Reply:
Done.